cloud/scope/powervs_cluster_test.go

```diff
@@ -3832,7 +3832,659 @@ func TestCreateVPCSubnet(t *testing.T) {
 		g.Expect(err).ToNot(BeNil())
 	})
 }
+func TestPowerVSDeleteLoadBalancer(t *testing.T) {
+	var (
+		mockVpc  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockVpc = mock.NewMockVpc(mockCtrl)
+	}
 
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	powervsClusterScope := func() *PowerVSClusterScope {
+		return &PowerVSClusterScope{
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					ServiceInstance: &infrav1beta2.ResourceReference{
+						ID: ptr.To("serviceInstanceID"),
+					},
+					LoadBalancers: map[string]infrav1beta2.VPCLoadBalancerStatus{
+						"lb": {
+							ID:                ptr.To("lb-id"),
+							ControllerCreated: ptr.To(true),
+						},
+					},
+				},
+			},
+		}
+	}
+
+	t.Run("When load balancer is not found", func(*testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+		clusterScope := powervsClusterScope()
+		mockVpc.EXPECT().GetLoadBalancer(gomock.Any()).Return(nil, &core.DetailedResponse{StatusCode: 404}, errors.New("not found"))
+		clusterScope.IBMVPCClient = mockVpc
+		requeue, err := clusterScope.DeleteLoadBalancer()
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+	t.Run("When IBMVPCClient DeleteLoadBalancer returns error", func(*testing.T) {
```

Comment:
```suggestion
	t.Run("Whe DeleteLoadBalancer returns error", func(*testing.T) {
```

Reply:
I think we can remove IBMVPCClient here and at other relevant places.

Reply:
Done.Removed it.